src/wifinode.cpp

```diff
@@ -52,3 +54,18 @@ void WifiNode::sendFragmentationNeeded(ipv4_headers_t* sourcePacket) {
   pbuf_free(pbuff);
   raw_remove(pcb);
 }
+
+void WifiNode::sendPacket(ipv4_headers_t* headers,
+                          void* payload,
+                          size_t length) {
+  struct raw_pcb* protocolControlBlock = raw_new(headers->protocol);
+  uint16_t* data = (uint16_t*)malloc(length);
+  memcpy(data, payload, length);
+  struct pbuf* pbuff = pbuf_alloc(PBUF_IP, *data, PBUF_RAM);
```

Comment:
Are you sure `pbuf_alloc` is being called with correct arguments?

Reply:
Now we are passing the correct arguments int routeToInternet of gateway.cpp and passing the correct arguments of in pbuff_alloc.